### DISCUSSION TITLE
事件选项出错

### DISCUSSION THREAD

**Discussion by liufuyoucong** [NateScarlet/auto-derby#200] [Q&A] [answered]
一到事件选择就会报错
INFO  [2021-10-06 20:34:26]:__main__:23: auto_derby: 1.8.1 (2021-10-01 11:55:57)
DEBUG [2021-10-06 20:34:26]:auto_derby.plugin:43: loaded: afk, auto_crane, example_custom_context, example_custom_race_score, example_custom_training_partner_score, example_custom_training_score, example_power_level5, example_smart_falcon, hello_world, less_op, limited_sale_buy_everything, limited_sale_buy_first_3, limited_sale_ignore, limited_sale_close, more_g1, prefer_g1, no_event_prompt, no_ocr_prompt, pause_before_command, pause_before_race_continue, SSR樫本理子, SSR駿川たづな, umapyoi, use_legacy_screenshot
INFO  [2021-10-06 20:34:26]:auto_derby.plugin:48: installed: no_ocr_prompt
INFO  [2021-10-06 20:34:26]:auto_derby.window:82: width=540 height=960
INFO  [2021-10-06 20:34:26]:auto_derby.window:84: already in wanted size
DEBUG [2021-10-06 20:34:27]:auto_derby.template:44: screenshot
DEBUG [2021-10-06 20:34:27]:auto_derby.template:58: load: connecting.pos.png
DEBUG [2021-10-06 20:34:27]:auto_derby.template:58: load: connecting.png
DEBUG [2021-10-06 20:34:27]:auto_derby.template:176: not match: tmpl=tmpl<connecting.png>, pos=(391, 55), similarity=0.184
DEBUG [2021-10-06 20:34:27]:auto_derby.template:58: load: retry_button.pos.png
DEBUG [2021-10-06 20:34:27]:auto_derby.template:75: can not load: retry_button.pos.png: [Errno 2] No such file or directory: 'D:\\auto-derby-master\\auto-derby-master\\auto_derby\\templates\\retry_button.pos.png'
DEBUG [2021-10-06 20:34:27]:auto_derby.template:58: load: retry_button.png
DEBUG [2021-10-06 20:34:27]:auto_derby.template:176: not match: tmpl=tmpl<retry_button.png>, pos=(0, 461), similarity=0.390
DEBUG [2021-10-06 20:34:27]:auto_derby.template:58: load: single_mode_command_training.pos.png
DEBUG [2021-10-06 20:34:27]:auto_derby.template:58: load: single_mode_command_training.png
DEBUG [2021-10-06 20:34:27]:auto_derby.template:176: not match: tmpl=tmpl<single_mode_command_training.png>, pos=(201, 759), similarity=0.210
DEBUG [2021-10-06 20:34:27]:auto_derby.template:58: load: single_mode_fans_not_enough.pos.png
DEBUG [2021-10-06 20:34:27]:auto_derby.template:58: load: single_mode_fans_not_enough.png
DEBUG [2021-10-06 20:34:27]:auto_derby.template:176: not match: tmpl=tmpl<single_mode_fans_not_enough.png>, pos=(263, 232), similarity=0.054
DEBUG [2021-10-06 20:34:27]:auto_derby.template:58: load: single_mode_target_race_no_permission.pos.png
DEBUG [2021-10-06 20:34:27]:auto_derby.template:58: load: single_mode_target_race_no_permission.png
DEBUG [2021-10-06 20:34:27]:auto_derby.template:176: not match: tmpl=tmpl<single_mode_target_race_no_permission.png>, pos=(155, 251), similarity=-0.082
DEBUG [2021-10-06 20:34:27]:auto_derby.template:58: load: single_mode_target_unfinished.pos.png
DEBUG [2021-10-06 20:34:27]:auto_derby.template:58: load: single_mode_target_unfinished.png
DEBUG [2021-10-06 20:34:27]:auto_derby.template:176: not match: tmpl=tmpl<single_mode_target_unfinished.png>, pos=(164, 248), similarity=0.102
DEBUG [2021-10-06 20:34:27]:auto_derby.template:58: load: single_mode_finish_button.pos.png
DEBUG [2021-10-06 20:34:27]:auto_derby.template:58: load: single_mode_finish_button.png
DEBUG [2021-10-06 20:34:27]:auto_derby.template:176: not match: tmpl=tmpl<single_mode_finish_button.png>, pos=(328, 811), similarity=0.023
DEBUG [2021-10-06 20:34:27]:auto_derby.template:58: load: single_mode_formal_race_banner.pos.png
DEBUG [2021-10-06 20:34:27]:auto_derby.template:58: load: single_mode_formal_race_banner.png
DEBUG [2021-10-06 20:34:27]:auto_derby.template:176: not match: tmpl=tmpl<single_mode_formal_race_banner.png>, pos=(387, 750), similarity=0.249
DEBUG [2021-10-06 20:34:27]:auto_derby.template:58: load: single_mode_race_next_button.pos.png
DEBUG [2021-10-06 20:34:27]:auto_derby.template:75: can not load: single_mode_race_next_button.pos.png: [Errno 2] No such file or directory: 'D:\\auto-derby-master\\auto-derby-master\\auto_derby\\templates\\single_mode_race_next_button.pos.png'
DEBUG [2021-10-06 20:34:27]:auto_derby.template:58: load: single_mode_race_next_button.png
DEBUG [2021-10-06 20:34:27]:auto_derby.template:176: not match: tmpl=tmpl<single_mode_race_next_button.png>, pos=(10, 472), similarity=0.464
DEBUG [2021-10-06 20:34:27]:auto_derby.template:58: load: single_mode_option1.pos.png
DEBUG [2021-10-06 20:34:27]:auto_derby.template:58: load: single_mode_option1.png
DEBUG [2021-10-06 20:34:27]:auto_derby.template:121: lightness match: tmpl=tmpl<single_mode_option1.png>, similarity=1.000
INFO  [2021-10-06 20:34:27]:auto_derby.template:180: match: tmpl=tmpl<single_mode_option1.png>, pos=(24, 525), similarity=1.00
DEBUG [2021-10-06 20:34:28]:auto_derby.template:44: screenshot
ERROR [2021-10-06 20:34:28]:__main__:123: unexpected exception
Traceback (most recent call last):
  File "D:\auto-derby-master\auto-derby-master\auto_derby\__main__.py", line 119, in <module>
    main()
  File "D:\auto-derby-master\auto-derby-master\auto_derby\__main__.py", line 90, in main
    job()
  File "D:\auto-derby-master\auto-derby-master\auto_derby\jobs\nurturing.py", line 222, in nurturing
    spec[_spec_key(tmpl)](ac)
  File "D:\auto-derby-master\auto-derby-master\auto_derby\jobs\nurturing.py", line 121, in _ac_handle_option
    _handle_option()
  File "D:\auto-derby-master\auto-derby-master\auto_derby\jobs\nurturing.py", line 35, in _handle_option
    ans = event.get_choice(template.screenshot(max_age=0))
  File "D:\auto-derby-master\auto-derby-master\auto_derby\single_mode\event.py", line 133, in get_choice
    reload_on_demand()
  File "D:\auto-derby-master\auto-derby-master\auto_derby\single_mode\event.py", line 82, in reload_on_demand
    reload()
  File "D:\auto-derby-master\auto-derby-master\auto_derby\single_mode\event.py", line 74, in reload
    g.choices = dict((k, int(v)) for k, v in csv.reader(f))
  File "D:\auto-derby-master\auto-derby-master\auto_derby\single_mode\event.py", line 74, in <genexpr>
    g.choices = dict((k, int(v)) for k, v in csv.reader(f))
ValueError: invalid literal for int() with base 10: 'ク
[Uploading debug.zip…]()
'

**Comment by liufuyoucong** [comment]
[debug.zip](https://github.com/NateScarlet/auto-derby/files/7293844/debug.zip)

**Comment by NateScarlet** [comment, marked as answer]
你应该是把 ocr_labels.csv 命名为 single_mode_choices.csv 了

删除你 single_mode_choices.csv 中的非法条目后重试

**Reply by liufuyoucong** [reply]
搞好了，谢谢
